src/appleseed/renderer/modeling/project/projectfileupdater.cpp

```diff
@@ -2109,12 +2110,19 @@ namespace
                         const uint64 max_samples = pfr.strings().get<uint64>("max_samples");
                         pfr.strings().remove("max_samples");
 
-                        Frame* frame = m_project.get_frame();
-                        if (frame)
+                        if (max_samples < numeric_limits<uint64>::max())
                         {
-                            // If max samples was previously set then preserve the nearest max average spp count.
-                            pfr.strings().insert(
-                                "max_average_spp", static_cast<uint64>(ceil(max_samples / frame->get_crop_window().volume())));
+                            Frame* frame = m_project.get_frame();
+                            if (frame)
+                            {
+                                // If max samples was previously set then preserve the nearest max average spp count.
```

Comment:
Maybe warn the user when this happen ?

Reply:
I missed that comment. Maybe it's a good idea yes.